packages/fcl/src/exec/query.js

```diff
@@ -43,25 +49,55 @@ import {normalizeArgs} from "./utils/normalize-args"
 export async function query(opts = {}) {
   await preQuery(opts)
 
-  // prettier-ignore
-  return sdk.send([
-    sdk.script(opts.cadence),
-    sdk.args(normalizeArgs(opts.args || [])),
-    opts.limit && typeof opts.limit === "number" && sdk.limit(opts.limit)
-  ]).then(sdk.decode)
+  if (isString(opts?.template)) {
```

Comment:
lines 52-70 are used in `mutate` as well.
Could add as `validateTemplate()` utils

`opt?.template && validateTemplate(opts.template)`

Reply:
Good call, added